Add building via conan

lo1ol opened the pull request:
Hi! 

My team using qbs as main build system in our projects. Recently we decided to pack all our dependencies via conan package manager. Qbs is also our dependency, so, we decided to pack it too.

I wrote conanfile.py for your project, which allows to build qbs with qt downloaded from conancenter and after building pack it to conan package. This package may be used as building tool in another projects:
```bash
conan install qbs/1.23.0@_/_ -g virtualenv -if install
source install/activate.sh
qbs --version
source install/deactivate.sh
```

I already manually tested using this package on Linux, Windows and Macos system, but think that will be better automate this process.

Except adding conanfile, I also 
1. Fix some problem with 'd' suffix of debug plugins on windows
2. Enable to specify user settings dir via environment variable. I think this feature will be useful for using packed qbs

What do you think about that?

lo1ol committed:
Add ability build qbs via conan

Change-Id: I84fb9b8aa142d6fa939befacc18fe901a00470ad

lo1ol committed:
Add 'd' suffix for debug builds of plugins on windows

Otherwise we get an error while build debug version of qbs

Change-Id: I30b0ff629800b5404ab0dc4b650e068978c87053

lo1ol committed:
Allow to get qbs settings dir from env var

This feature will be useful for using qbs conan package at virtual
environment

Change-Id: I0d903d9696edcc9b339226deb2cb38de89e319f5

mabean commented on conanfile.py:
```diff
@@ -0,0 +1,94 @@
+            self.tool_requires("jom/1.1.3")
+
+    def source(self):
+        self.run(f"git clone -b v{self.version} --recursive -- https://github.com/qbs/qbs {self.source_folder}")
```
It's better to use `tools.Git` I suppose. The best way would be to download archives from qt mirrors and check the sha256 like it's getting done in most of [conan-center-index](https://github.com/conan-io/conan-center-index) recipes using conandata.yml

lo1ol replied:
It sounds really better, thanks)

mabean commented on conanfile.py:
```diff
@@ -0,0 +1,94 @@
+        self.output.info(f"Appending PATH environment variable: {bindir}")
+        self.env_info.PATH.append(bindir)
+
+        if self.options.shared:
```
It's looks like Qt isn't getting bundled in qbs package, that isn't good. AFAIK there is no flag to do it using qmake. There are `modules.qbsbuildconfig.useBundledQtScript:true` and `modules.qbsbuildconfig.enableBundledQt:true` flags to allow you to do this to build qbs with qbs.

lo1ol replied:
Ok, I will try to add this to qmake

lo1ol commented:
Thank you for reply and tips! I don't know about conan.center repo. I will try to add qbs conanfile there.

lo1ol commented:
For another two commits it will be better to create MRs?

lo1ol committed:
Add ability to bundle qt to build and install folder via qmake

This feature will be used for creation qbs conan package:
https://github.com/conan-io/conan-center-index/pull/11592

Change-Id: I8d41544252048300641398446c2f1725156b020e

mabean commented:
> For another two commits it will be better to create MRs?

So, I suppose you want to contribute your patches via gerrit to qbs and contribute conanfile with the same patches for now to conan-center. You can't contribute to qbs via github. Talking about conanfile, now it looks fine for me.

lo1ol commented:
https://codereview.qt-project.org/c/qbs/qbs/+/421320
https://codereview.qt-project.org/c/qbs/qbs/+/421321
https://codereview.qt-project.org/c/qbs/qbs/+/421322

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/qbs/qbs?pullRequest=8) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/qbs/qbs?pullRequest=8) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/qbs/qbs?pullRequest=8) it.</sub>
